File: Projetos/Alcoolougasolina/app/src/main/java/com/estudosandroid/vinibza/alcoolougasolina/MainActivity.java
package com.estudosandroid.vinibza.alcoolougasolina;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import com.google.android.material.textfield.TextInputEditText;

public class MainActivity extends AppCompatActivity {
//Componentes de tela
    private TextInputEditText editAlcool, editGasolina;
    private Button calcular;
    private TextView resultado;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        editAlcool    = findViewById(R.id.editAlcool);
        editGasolina  = findViewById(R.id.editGasolina);
        calcular  = findViewById(R.id.btnCalcular);
        resultado = findViewById(R.id.txtResultado);
    }

    public void calcularPreco(View view){
        String precoAlcool = editAlcool.getText().toString();
        String precoGasolina = editGasolina.getText().toString();

        //Validar campos
        boolean camposValidados = validarCampos(precoAlcool, precoGasolina);
        if (camposValidados == true){
            // Converter String para Double
            Double valorAlcool = Double.parseDouble(precoAlcool);
            Double valorGasolina = Double.parseDouble(precoGasolina);
            Double resultadoValores = valorAlcool / valorGasolina;

          if (resultadoValores >= 0.7){
              resultado.setText("A Gasolina é a melhor opção ");
             }else {
               resultado.setText("O Alcool é a melhor opção. "); }
        }else {
            resultado.setText("Por favor, insira o valor nos campos.");
        }

    }

    public boolean validarCampos(String pAlcool, String pGasolina){
         Boolean camposValidados = true;
         /*
         * Realizamos uma validação de null caso não conseguimos recuperar o valor que o usuario digitar
         * ou então não configurasemos o pAlcool por exemplo. E o Equal verifica se a condição é verdadeira
         * ou falsa, nesse caso se o campo estiver vazio.
         * */
         if (pAlcool == null || pAlcool.equals("")){
             camposValidados = false;
         }
         else if(pGasolina == null || pGasolina.equals("")){
             camposValidados = false;
         }
         return camposValidados;
    }

}

File: Projetos/Livro Kotlin/CalculoAposentadoria/app/src/main/java/livrokotlin/com/br/MainActivity.kt
package livrokotlin.com.br

import android.os.Bundle
import android.widget.*
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val spn_sexo = findViewById<Spinner>(R.id.spn_sexo)
        val edt_idade = findViewById<EditText>(R.id.edt_idade)
        val btn_calcular = findViewById<Button>(R.id.btn_calcular)
        val txt_resultado = findViewById<TextView>(R.id.txt_resultado)

        spn_sexo.adapter = ArrayAdapter<String>(this, android.R.layout.simple_spinner_dropdown_item,
        listOf("Masculino", "Feminino"))

        btn_calcular.setOnClickListener {

            //Faz com que o retorno do spn seja do tipo string
            val sexo = spn_sexo.selectedItem as String

            //recuperar a idade do usuário
            val idade = edt_idade.text.toString().toInt()

            var resultado = 0
            if (sexo == "Masculino") {
                resultado = 65 - idade
            }else{
                resultado = 60 - idade
            }

            //Atualizando o resultado de acordo com o fornecido.
            txt_resultado.text = " Faltam $resultado anos para você se aposentar."
        }
    }
}